podaac/swodlr/cli/__main__.py

```diff
@@ -0,0 +1,157 @@
+        '--limit',
+        help='The number of results to return',
+        default=10
+    )
+    
+    # -- invalidate-product
+    invalidate_product_parser = subparsers.add_parser(
+        'invalidate-product',
+        help='Invalidate a product by product id; allows the product to be ' +
+        'regenerated regardless of current product status by transitioning ' +
+        'the product to UNAVAILABLE'
+    )
+    invalidate_product_parser.set_defaults(func=invalidate_product)
+    invalidate_product_parser.add_argument('id')
+    
+    # -- search-products
+    search_products_parser = subparsers.add_parser(
+        'search-products',
+        help='Search for existing products in the cache without generating ' +
+        'any new products'
+    )
+    search_products_parser.set_defaults(func=search_products)
+    search_products_parser.add_argument('--cycle')
+    search_products_parser.add_argument('--pass')
+    search_products_parser.add_argument('--scene')
+    search_products_parser.add_argument('--output-granule-extent-flag')
+    search_products_parser.add_argument('--output-sampling-grid-type')
+    search_products_parser.add_argument('--raster-resolution')
+    search_products_parser.add_argument('--utm-zone-adjust')
+    search_products_parser.add_argument('--mgrs-band-adjust')
+    search_products_parser.add_argument(
+        '--after-id',
+        help='Used for pagination; the last product id from the last page'
+    )
+    invalidate_product_parser.add_argument(
+        '--limit',
+        help='The number of results to return',
+        default=10
+    )
+    
+    args = parser.parse_args()
+    
+    if not hasattr(args, 'func'):
+        parser.print_help()
+        return
+
+    # Configure client before passing to function
+    token = earthaccessapi.login().token
+    transport = AIOHTTPTransport(
+        url=_get_graphql_url(args.env),
+        headers={'Authorization': f'Bearer {token['access_token']}'}
+    )
+    session = gql.Client(transport=transport)
+    func = args.func
+        
+    # Remove misc params and convert to dict
+    args = vars(args)
+    for key in ['func', 'env']:
+        if key in args:
+            del args[key]
+
+    func(session, args)
+
+
+def get_users_products(session, args):
+    query = _load_graphql_query('get_users_products')
+    results = session.execute(query, variable_values=args)
+    print(json.dumps(results, indent=2))
+
+
+def invalidate_product(session, args):
+    query = _load_graphql_query('invalidate_product')
+    results = session.execute(query, variable_values=args)
+    print(json.dumps(results, indent=2))
+
+
+def search_products(session, args):
+     query = _load_graphql_query('search_products')
+     results = session.execute(query, variable_values=args)
+     print(json.dumps(results, indent=2))
+
+
+def _load_graphql_query(name):
+    path = Path(__file__, '..', 'graphql-documents', f'{name}.graphql').resolve()
```

Comment:
Can you convert this to using https://docs.python.org/3/library/importlib.resources.html#module-importlib.resources instead?